### DISCUSSION TITLE
How to retrigger a defined query (was defined using useQuery), without a queryFn?

### DISCUSSION THREAD

**Discussion by Jurek-Raben** [TanStack/query#5964] [Q&A]
I would like to await a query, which already was defined before in a useQuery, but retrigger it without the need to define a new queryFn. Instead, it should retrigger the one before... is this possible?

This should happen within the react router v6 loader method, here some pseudo code:

```
const loader = (queryClient) => {
  await queryClient.retriggerQuery({
    queries: [
      {
        queryKey: [key1]
      },
      {
        queryKey: [key2]
      },
    ]
  });
  return null;
}
```

**Comment by Birkbjo** [comment]
Not sure how your keys are structured, are they related? Eg. maybe they begin with a common string or object?

Anyway you can use or `refetchQueries` or `invalidateQueries` depending on your usecase.
```ts
await queryClient.refetchQueries({ queryKey: [key] })

// or
await queryClient.invalidateQueries({ queryKey: [key] }) //may need do specify refetchType depending on your usecase

``` 

[QueryClient docs](https://tanstack.com/query/v3/docs/react/reference/QueryClient#queryclientrefetchqueries)
